src/dynamics/video_player/player/player.js

```diff
@@ -614,62 +612,64 @@ Scoped.define("module:VideoPlayer.Dynamics.Player", [
                             return false;
                         }
                     },
-                    "containerSizingStyles:aspect_ratio,height,width,floating_height,floating_width,floating_top,floating_right,floating_bottom,floating_left,is_floating,adsinitialized,states.hiddenelement": function(aspectRatio, height, width, floatingHeight, floatingWidth, floatingTop, floatingRight, floatingBottom, floatingLeft, isFloating, adsInitialized, hiddenelement) {
+                    "hideplayer:adshassource,adsinitialized,hidden,hidebeforeadstarts": function(
+                        adshassource,
+                        adsinitialized,
+                        hidden,
+                        hidebeforeadstarts
+                    ) {
+                        if (hidden) return hidden;
+                        if (hidebeforeadstarts && adshassource) return !adsinitialized;
+                        return false;
+                    },
+                    "containerSizingStyles:aspect_ratio,height,width,mobileviewport,is_floating,hideplayer,corner,floatingoptions.floatingonly": function(
+                        aspectRatio,
+                        height,
+                        width,
+                        mobileviewport,
+                        isFloating,
+                        hidden,
+                        corner,
+                        floatingonly
+                    ) {
                         var containerStyles, styles, calculated;
                         styles = {
                             aspectRatio: aspectRatio
                         };
                         if (height) styles.height = isNaN(height) ? height : parseFloat(height).toFixed(2) + "px";
                         if (width) styles.width = isNaN(width) ? width : parseFloat(width).toFixed(2) + "px";
-                        containerStyles = styles;
-                        if (isFloating && !this.get("fullscreened")) {
+                        containerStyles = floatingonly ? {} : Objs.extend({}, styles);
+                        if (corner) styles.borderRadius = corner;
+                        if (isFloating) {
                             calculated = this.__calculateFloatingDimensions();
 
```

Comment:
It's not critical but will be better to predefined variables. 
`let floatingTop, floatingBottom, floatingRight, floatingLeft, floatingWidth, floatingHeight;`